language.md

```diff
@@ -0,0 +1,543 @@
+If you don't pass the stdout channel to a function, there's no way for it to print anything.
+This is especially useful for more "powerful" capabilities like accessing the file system or network:
+When using a package, without reading its source code, you can be confident that it won't delete your files under some special circumstances.
+
+If a function expects a stdout channel, there's no way it can tell if you gave it another channel that you just created.
+You could for example process the output of the function, filter some information out, and forward the rest to the real stdout channel.
+
+## Interoperability with other languages
+
+Candy has no plans to directly support Foreign Function Interfaces (FFI) to communicate with other code.
+The reason is that doing so inherently breaks the isolation of code.
+
+Depending on the use case, we offer two alternative options:
+
+### Add to the Environment
+
+If you develop for a new platform or want to enable more functionality in the native platform, we will have some way of plugging a new part of native code into the runtime that can make its own capabilities available on the environment passed to `main`.
+
+For example, on a microcontroller, the stdout capability doesn't make sense.
+Instead, you might wave a pin capability that allows you to modify the voltage of the hardware pins.
+
+### Contain Pure Code
+
+If you want to use existing code that implements pure functions, it can make sense to compile the existing code into WebAssembly (WASM).
+You can put the resulting WASM module in your Candy module hierarchy, call `useAsset` with that file, and pass the data to a WASM runtime that we'll write in Candy when we get to it.
+
+This approach moves the native code entirely into the Candy domain, so the Candy compiler can also reason about the WASM code.
+For example, if you call a function of your WASM module only with inputs known at compile-time, the Candy compiler may execute those calls directly and not include the original WASM code in the binary at all.
+
+## Deploying code
+
+This chapter is especially experimental and spitball-y.
+
+Similar to how Zig build scripts work, we may support having a `build.candy` file that contains information about how to compile and optimize your code.
+
+Next to the interpreted VM, we plan to compile to LLVM or WASM.
+
+One idea we had is to let you provide a custom scoring function in the build script instead of having binary options like "optimize for speed" or "optimize for performance".
+This scoring function could get used by the optimizer to choose which paths to take.
+For example, you could formulate a build where you're okay with a resulting binary blowup of 1 KiB per 10 ms of saved time in some annotated performance-crititcal section.
+Or when developing for an embedded device with limited storage capacity, you might want to generate a binary that fits in the limit but is otherwise as fast as possible.
```

Comment:
```suggestion
Or, when developing for an embedded device with limited storage capacity, you might want to generate a binary that fits within the limit but is otherwise as fast as possible.
```